### DISCUSSION TITLE
后端服务高可用，当服务a1进行升级时，如何避免rocketmq消息下发到升级节点？

### DISCUSSION THREAD

**Discussion by djl-kara** [apache/rocketmq#8175] [Q&A]
后端服务高可用a1和a2，当服务a1进行升级时
调整a1服务权重调整为0，进行服务升级时，这时候如果rocketmq对a1服务下发了消息，但是a1服务处于宕机状态无法消费，此时消息消费失败，然后进入重试机制，
该消息是否会重新分发给其他节点，如a2？
还是说，该消息会等a1服务恢复正常后才重新进行消费？

**Comment by cserwen** [comment]
a1 升级期间，消费者理应进行 shutdown ，服务端不会再向 shutdown 的消费者发送消息

**Reply by cserwen** [reply]
即使没有 shutdown，进程停止，服务端也会认为这个消费者下线了，不会再下发消息
